js/src/contract/item.ts

```diff
@@ -1,417 +1,37 @@
-  newItemTokenHolder: web3.PublicKey | null;
-  parentMint: web3.PublicKey | null;
-  metadataUpdateAuthority: web3.PublicKey | null;
-}
-
-export interface AddCraftItemToEscrowAccounts {
-  itemClassMint: web3.PublicKey;
-  newItemToken: web3.PublicKey | null;
-  newItemTokenHolder: web3.PublicKey | null;
-  craftItemTokenMint: web3.PublicKey | null;
-  parentMint: web3.PublicKey | null;
-  metadataUpdateAuthority: web3.PublicKey | null;
-}
-
-export interface UpdateValidForUseIfWarmupPassedAccounts {}
-
-export interface RemoveCraftItemFromEscrowAccounts {
-  itemClassMint: web3.PublicKey;
-  newItemToken: web3.PublicKey | null;
-  newItemTokenHolder: web3.PublicKey | null;
-  parentMint: web3.PublicKey | null;
-  metadataUpdateAuthority: web3.PublicKey | null;
-}
-
-export interface CompleteItemEscrowBuildPhaseAccounts {
-  itemClassMint: web3.PublicKey;
-  newItemMint: web3.PublicKey;
-  newItemToken: web3.PublicKey | null;
-  newItemTokenHolder: web3.PublicKey | null;
-  parentMint: web3.PublicKey | null;
-  metadataUpdateAuthority: web3.PublicKey | null;
-}
-
-export interface DeactivateItemEscrowAccounts {}
-
-export interface DrainItemEscrowAccounts {
-  originator: web3.PublicKey | null;
-}
-
-export interface UpdateItemClassAccounts {
-  itemMint: web3.PublicKey;
-  parent: web3.PublicKey | null;
-  parentMint: web3.PublicKey | null;
-  metadataUpdateAuthority: web3.PublicKey | null;
-}
-
-export interface UpdateItemAccounts {}
-
-export interface StartItemEscrowBuildPhaseAccounts {
-  itemClassMint: web3.PublicKey;
-  newItemToken: web3.PublicKey | null;
-  newItemTokenHolder: web3.PublicKey | null;
-  parentMint: web3.PublicKey | null;
-  metadataUpdateAuthority: web3.PublicKey | null;
-}
-
-export interface BeginItemActivationAdditionalArgs {}
-
-export interface EndItemActivationAdditionalArgs {}
-
-export interface CreateItemClassAdditionalArgs {}
-
-export interface UpdateItemClassAdditionalArgs {
-  permissionless: boolean;
-}
-
-export interface CreateItemEscrowAdditionalArgs {}
-
-export interface StartItemEscrowBuildPhaseAdditionalArgs {}
-export interface CompleteItemEscrowBuildPhaseAdditionalArgs {}
-
-export interface UpdateItemAdditionalArgs {}
-export interface AddCraftItemToEscrowAdditionalArgs {}
-export interface RemoveCraftItemFromEscrowAdditionalArgs {}
-
-export interface DeactivateItemEscrowAdditionalArgs {}
-
-export interface DrainItemEscrowAdditionalArgs {}
-
-export interface UpdateValidForUseIfWarmupPassedAdditionalArgs {}
+import { ITEM_ID } from "../constants/programIds";
+import * as ItemInstruction from "../instructions/item";
+import { decodeItemClass, ItemClass } from "../state/item";
+import { getAtaForMint, getItemPDA } from "../utils/pda";
 
-export class ItemProgram {
-  id: web3.PublicKey;
-  program: Program;
+export class ItemProgram extends Program.Program {
+  declare instruction: ItemInstruction.Instruction;
+  PROGRAM_ID = ITEM_ID;
+  // QUESTION: Should we specify a PREFIX here?
```

Comment:
Sorry @long-banana, one more question for you

Reply:
Yup. As a static variable.

https://github.com/raindrop-studios/rain-redemptions/blob/3436c45c057dea87136444c3ff97357442278c1a/app/lib/src/programs/redemption.ts#L11